### PR TITLE
fix: sync failed after lost data

### PR DESCRIPTION
**Pull request by andyoknen**
## Standards checklist:

- [x] The PR title is descriptive.
- [x] The PR doesn't replicate another PR which is already open.
- [x] The PR has self-explained commits history.

- [x] The code is mine or it's from somewhere with an Apache-2.0 compatible license.
- [x] The code is efficient, to the best of my ability, and does not waste computer resources.
- [x] The code is stable and I have tested it myself, to the best of my abilities.
- [x] If the code introduces new classes, methods, I provide a valid use case for all of them.

## Changes:

- A lot of fixes aimed at stabilizing the work of the WAL log

### CODE REVIEW

**Review by only1question — APPROVED** [review]

**Review by zedxter — APPROVED** [review]

**Review by lostystyg — COMMENTED** [review]
61facd0ad4f56b37310dc40d2f23e3d4e78b927a That was previously more correct, isn't it?

File: src/pocketdb/services/WebPostProcessing.cpp [hunk 1]
```diff
@@ -47,38 +42,11 @@ namespace PocketServices
...
-                    webRepoInst->UpsertBarteronAccounts(queueRecord.BlockHeight);
-                    webRepoInst->UpsertBarteronOffers(queueRecord.BlockHeight);
-                    break;
-                    // queueRecord.Height % N == 0 -> actions
-                }
-                default:
-                    break;
-            }
+            if (!ProcessNextHeight())
+                std::this_thread::sleep_for(1000ms);
```
Comment (lostystyg) [inline src/pocketdb/services/WebPostProcessing.cpp]: Use `UninterruptibleSleep` for coherence with other code

File: src/pocketdb/repositories/CheckpointRepository.cpp [hunk 2]
```diff
@@ -361,6 +361,1138 @@ namespace PocketDb
         _socialCheckpoints.emplace_back("ece0d09e1e89e15582f85e90a1d4df2e5efa9d96e3a2d15dcc9648b4db814ac7", 300, 4);
         _socialCheckpoints.emplace_back("f9e1d693b97da9688b9e1353a334aa167d5134a087e690bd462bb9157e000f69", 300, 4);
         _socialCheckpoints.emplace_back("6fcad0f123d4781dd846076a705aadd9c4f347920a76d88dfbe9905777afb8ae", 303, 8);
+        _socialCheckpoints.emplace_back("019d34809560f74e7ba065b1943eb241d29ba5427caefdf922774610c07521af", 100, 6);
```
Comment (lostystyg) [inline src/pocketdb/repositories/CheckpointRepository.cpp:364]: Please describe what is this

**Comment by andyoknen** [comment]
> [61facd0](https://github.com/pocketnetteam/pocketnet.core/commit/61facd0ad4f56b37310dc40d2f23e3d4e78b927a) That was previously more correct, isn't it?

Before the correction, a situation is created when the web is ahead of the main database (for example, during reindexation), then 0 is always listed in the statistics. A mechanism is also needed in the future to rebuild the web after rollbacks

File: src/pocketdb/repositories/CheckpointRepository.cpp (hunk 2 above)
Comment (andyoknen) [reply src/pocketdb/repositories/CheckpointRepository.cpp:364]: For users, we changed the rules of name processing or the order of editing. The old rules were eliminated to make the code easier and checkpoints were added to ensure synchronization. Validation of the data is maintained at the OP_RETURN validation level

**Review by lostystyg — APPROVED** [review]

**Pull request merged**